xdsl/backend/riscv/lowering/rv32_arith_lowering.py

```diff
@@ -0,0 +1,436 @@
+
+
+class LowerArithMinSI(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.MinSI, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("MinSI is not supported")
+
+
+class LowerArithMaxSI(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.MaxSI, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("MaxSI is not supported")
+
+
+class LowerArithMinUI(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.MinUI, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("MinUI is not supported")
+
+
+class LowerArithMaxUI(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.MaxUI, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("MaxUI is not supported")
+
+
+class LowerArithCmpi(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.Cmpi, rewriter: PatternRewriter) -> None:
+        # based on https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/RISCV/i32-icmp.ll
+        match op.predicate.value.data:
+            # eq
+            case 0:
+                xor_op = riscv.XorOp(op.lhs, op.rhs)
+                seqz_op = riscv.SltiuOp(xor_op, 1)
+                rewriter.replace_matched_op([xor_op, seqz_op])
+            # ne
+            case 1:
+                zero = riscv.GetRegisterOp(riscv.Registers.ZERO)
+                xor_op = riscv.XorOp(op.lhs, op.rhs)
+                snez_op = riscv.SltuOp(zero, xor_op)
+                rewriter.replace_matched_op([zero, xor_op, snez_op])
+                pass
+            # slt
+            case 2:
+                rewriter.replace_matched_op([riscv.SltOp(op.lhs, op.rhs)])
+            # sle
+            case 3:
+                slt = riscv.SltOp(op.lhs, op.rhs)
+                xori = riscv.XoriOp(slt, 1)
+                rewriter.replace_matched_op([slt, xori])
+            # ult
+            case 4:
+                rewriter.replace_matched_op([riscv.SltuOp(op.lhs, op.rhs)])
+            # ule
+            case 5:
+                sltu = riscv.SltuOp(op.lhs, op.rhs)
+                xori = riscv.XoriOp(sltu, 1)
+                rewriter.replace_matched_op([sltu, xori])
+            # ugt
+            case 6:
+                rewriter.replace_matched_op([riscv.SltuOp(op.rhs, op.lhs)])
+            # uge
+            case 7:
+                sltu = riscv.SltuOp(op.rhs, op.lhs)
+                xori = riscv.XoriOp(sltu, 1)
+                rewriter.replace_matched_op([sltu, xori])
+            case _:
+                raise NotImplementedError("Cmpi predicate not supported")
+
+
+class LowerArithSelect(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: arith.Select, rewriter: PatternRewriter) -> None:
+        raise NotImplementedError("Select is not supported")
```

Comment:
Feels like ideally we should have a python test that tests for this exception and all of the others

Reply:
I'm not sure how useful it is right now since the coverage of these ops might be improved very soon.

Reply:
Good point!